### PR TITLE
Improve beforeDelete event error handling

### PR DESCRIPTION
**Pull request by nicosp**
Return false instead of throwing an exception if save fails to allow other listeners to abort the update with proper error handling.

### CODE REVIEW

**Comment by ADmad** [comment]
A new test case if required for this change.

**Comment by nicosp** [comment]
> A new test case if required for this change.

Test added

**Comment by ADmad** [comment]
The `@return true` in docblock needs to be changed to `@return bool`.

**Comment by nicosp** [comment]
> The `@return true` in docblock needs to be changed to `@return bool`.

Done

File: tests/TestCase/Model/Behavior/TrashBehaviorTest.php [hunk 1]
```diff
@@ -163,6 +163,33 @@ public function testBeforeDelete()
...
+
+        $this->Articles->getEventManager()->on(
+            'Model.beforeSave',
+            [],
+            function (Event $event, EntityInterface $entity, ArrayObject $options) {
+                $entity->setError('id', 'Save aborted');
+                $event->setResult(false);
+                $event->stopPropagation();
+            }
+
```
Comment (ADmad) [inline tests/TestCase/Model/Behavior/TrashBehaviorTest.php:183]:
```suggestion
```

File: tests/TestCase/Model/Behavior/TrashBehaviorTest.php [hunk 2]
```diff
@@ -163,6 +163,33 @@ public function testBeforeDelete()
...
+        );
+
+        $result = $this->Articles->delete($article);
+
+        $this->assertFalse($result);
+        $this->assertArrayHasKey('id', $article->getErrors());
+    }
+
```
Comment (ADmad) [inline tests/TestCase/Model/Behavior/TrashBehaviorTest.php:191]:
```suggestion
```

**Pull request merged**